src/main/java/com/cgm/infolab/controller/RoomApiController.java

```diff
@@ -0,0 +1,58 @@
+package com.cgm.infolab.controller;
+
+import com.cgm.infolab.db.model.RoomEntity;
+import com.cgm.infolab.db.repository.RoomRepository;
+import com.cgm.infolab.model.RoomDto;
+import com.cgm.infolab.service.RoomService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.security.Principal;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+public class RoomApiController {
+    private final RoomService roomService;
+
+    private final Logger log = LoggerFactory.getLogger(RoomApiController.class);
+
+    @Autowired
+    public RoomApiController(RoomService roomService) {
+        this.roomService = roomService;
+    }
+
+    // TODO: magari è necessario far si che le room prese siano soltanto quelle assegnate ad uno user,
+    //  in modo che uno user non possa vedere le room di qualcun altro.
+    //  Per farlo o si stabilisce una convenzione per capire se uno user è in una stanza,
+    //  o va aggiunta una relazione al db.
+    @GetMapping("/api/rooms")
+    public List<RoomDto> getAllRooms(@RequestParam(required = false) String date, Principal principal) {
+        List<RoomDto> roomDtos = new ArrayList<>();
+
+        List<RoomEntity> roomEntities = roomService.getRoomsAfterDate(fromStringToDate(date));
```

Comment:
Quasi... in realtà è tutto questo pezzo che andrebbe messo dentro al service. Da linea 37 a linea 47. E' tutta logica di business. Anche il metodo `fromStringToDate`